File: leetcode/169.majority-element.py
# @leet start
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        curr = nums[0]
        count = 0

        for num in nums:
            if num == curr:
                count += 1
            elif count:
                count -= 1
            else:
                curr = num
                count = 1

        return curr


# @leet end

File: leetcode/169.majority-element.ts
// @leet start
function majorityElement(nums: number[]): number {
  let curr = 0;
  let score = 0;

  for (const num of nums) {
    if (!score) {
      curr = num;
    }

    score += num === curr ? 1 : -1;
  }

  return curr;
}
// @leet end
